src/compiler/chainrules.jl

```diff
@@ -1,3 +1,8 @@
+@inline unthunk_tangent(x::AbstractThunk) = wrap_chainrules_output(unthunk(x))
+@inline unthunk_tangent(x::AbstractArray{<:AbstractThunk}) = map(unthunk_tangent, x)
+unthunk_tangent(d::IdDict) = IdDict([unthunk_tangent(k) => unthunk_tangent(v) for (k, v) in d])
```

Comment:
```suggestion
unthunk_tangent(d::IdDict) = IdDict([unthunk_tangent(k) => unthunk_tangent(v) for (k, v) in d])
@non_differentiable unthunk_tangent(::IdDict)
```
This would probably fix the "map is not defined on dictionaries" error, but unsure if it's sufficient to pass the nested params test.

Reply:
After some more testing, I suspect that Zygote is relying on the identity of the inner `IdDict` for parameter tracking. Breaking that identity with `unthunk_tangent(d::IdDict)` seems to cause propagation of null gradients at unexpected places.

Reply:
```suggestion
function ChainRulesCore.rrule(::typeof(unthunk_tangent), d::IdDict)
  unthunk_iddict_pullback(_) = (NoTangent(), ChainRulesCore.@not_implemented "unthunking IdDict")
  return d, unthunk_iddict_pullback
end
```
This seems to resolve the follow-up test failure and not cause any new ones, but can not 100% vouch for its correctness.